dsl/src/main/java/tech/pegasys/peeps/network/Network.java

```diff
@@ -254,7 +270,11 @@ public EthSigner addSigner(
                 .withContainerNetwork(subnet.network())
                 .withIpAddress(subnet.getAddressAndIncrement())
                 .withDownstream(downstream)
-                .withChainId(genesis.getConfig().getChainId())
+                .withChainId(
+                    genesisMap
+                        .get(Web3ProviderType.BESU)
+                        .getConfig()
+                        .getChainId()) // yeah, this is a bit of a hack.
```

Comment:
maybe it's ok for now? but yeah it is a bit hacky. we don't know if it's besu or that we are going to use the same chain id for every node.